be/src/exec/vectorized/chunks_sorter_topn.cpp

```diff
@@ -460,4 +464,28 @@ Status ChunksSorterTopn::_hybrid_sort_first_time(RuntimeState* state, Permutatio
     return Status::OK();
 }
 
+void ChunksSorterTopn::_rank_pruning() {
+    if (_topn_type != TTopNType::RANK) {
+        return;
+    }
+    if (_merged_segment.order_by_columns.empty()) {
+        return;
+    }
+    if (_merged_segment.chunk->num_rows() <= _get_number_of_rows_to_sort()) {
+        return;
+    }
+    const auto peer_group_start = static_cast<int>(_get_number_of_rows_to_sort()) - 1;
+    DCHECK_GE(peer_group_start, 0);
+    Tie tie(_merged_segment.chunk->num_rows(), 1);
+    vectorized::build_tie_for_columns(_merged_segment.order_by_columns, &tie);
+    const auto peer_group_end = SIMD::find_zero(tie, peer_group_start + 1);
+
+    for (auto& column : _merged_segment.chunk->columns()) {
```

Comment:
why not resize the chunk directly?

Reply:
There is no resize method of `vectorized::Chunk` 😂